File: firebase/SQLiteApplication/app/src/main/java/com/example/sqliteapplication/DBHelper.java
package com.example.sqliteapplication;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import java.util.ArrayList;
import java.util.List;


public class DBHelper extends SQLiteOpenHelper {

    private static final int DATABASE_VERSION = 1;
    private static final String DATABASE_NAME = "UserManager.db";
    private static final String TABLE_USER = "user";
    private static final String COLUMN_ID = "id";
    private static final String COLUMN_USER_NAME = "user_name";
    private static final String COLUMN_USER_ID = "user_id";
    private static final String COLUMN_USER_PASSWORD = "user_password";



    private final String DROP_USER_TABLE = "DROP TABLE IF EXISTS " + TABLE_USER;

    public DBHelper(Context context) {
        super(context, DATABASE_NAME, null, DATABASE_VERSION);
    }

    @Override
    public void onCreate(SQLiteDatabase db) {
        db.execSQL(CREATE_USER_TABLE);
    }
    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        db.execSQL(DROP_USER_TABLE);
        onCreate(db);
    }

    public void addUser(User user) {
        SQLiteDatabase db = this.getWritableDatabase();
        ContentValues values = new ContentValues();
        values.put(COLUMN_USER_NAME, user.getUsername());
        values.put(COLUMN_USER_ID, user.getUserID());
        values.put(COLUMN_USER_PASSWORD, user.getPassword());
        // Inserting Row
        db.insert(TABLE_USER, null, values);
        db.close();
    }

    public boolean checkUser(String userID, String password) {
        String[] columns = {
                COLUMN_USER_ID
        };
        SQLiteDatabase db = this.getReadableDatabase();
        String selection = COLUMN_USER_ID + " = ?" + " AND " + COLUMN_USER_PASSWORD + " = ?";
        String[] selectionArgs = {userID, password};

        Cursor cursor = db.query(TABLE_USER,
                columns, selection, selectionArgs, null, null, null);

        int cursorCount = cursor.getCount();
        cursor.close();
        db.close();
        return cursorCount > 0;
    }

    public List<User> getAllUser() {

        String[] columns = {
                COLUMN_ID,
                COLUMN_USER_ID,
                COLUMN_USER_NAME,
                COLUMN_USER_PASSWORD
        };

        String sortOrder =
                COLUMN_USER_NAME + " ASC";
        List<User> userList = new ArrayList<User>();
        SQLiteDatabase db = this.getReadableDatabase();

        Cursor cursor = db.query(TABLE_USER, columns, null, null,null,null, sortOrder);

        if (cursor.moveToFirst()) {
            do {
                User user = new User();
                user.setID(Integer.parseInt(cursor.getString(cursor.getColumnIndex(COLUMN_ID))));
                user.setUsername(cursor.getString(cursor.getColumnIndex(COLUMN_USER_NAME)));
                user.setUserID(cursor.getString(cursor.getColumnIndex(COLUMN_USER_ID)));
                user.setPassword(cursor.getString(cursor.getColumnIndex(COLUMN_USER_PASSWORD)));

                userList.add(user);
            } while (cursor.moveToNext());
        }
        cursor.close();
        db.close();
        return userList;
    }



    // create table sql query
    private final String CREATE_USER_TABLE = "CREATE TABLE " + TABLE_USER + "("
            + COLUMN_ID + " INTEGER PRIMARY KEY AUTOINCREMENT," + COLUMN_USER_NAME + " TEXT,"
            + COLUMN_USER_ID + " TEXT," + COLUMN_USER_PASSWORD + " TEXT" + ")";
}

File: services/DBServiseAplication/app/src/main/java/com/example/dbserviseaplication/DBHelper.kt
package com.example.dbserviseaplication

import android.app.Activity
import android.content.ContentValues
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper
import com.example.dbserviseaplication.models.model

class DBHelper (activity: Activity) :
SQLiteOpenHelper(activity, DATABASE_NAME, null, DATABASE_VERSION) {

    private val db: SQLiteDatabase = this.writableDatabase

    override fun onCreate(db: SQLiteDatabase?) {
        db!!.execSQL(model.TABLE_CREATE)
    }

    override fun onUpgrade(db: SQLiteDatabase?, oldVersion: Int, newVersion: Int) {
        db!!.execSQL("Drop table if exists ${model.TABLE_NAME}")
        onCreate(db)
    }


    fun insertPerson(name: String, password: String , email: String,phonenumber: String): Boolean {
        val cv = ContentValues()
        cv.put(model.COL_NAME, name)
        cv.put(model.COL_PASSWORD, password)
        cv.put(model.COL_EMAIL, email)
        cv.put(model.COL_PHONENUMBER, phonenumber)
        return db.insert(model.TABLE_NAME, null, cv) > 0
    }

    fun getAllPerson(): ArrayList<model> {
        val data = ArrayList<model>()
        val c =
            db.rawQuery("select * from ${model.TABLE_NAME} order by ${model.COL_ID} desc", null)
        c.moveToFirst()
        while (!c.isAfterLast) {
            val p = model(c.getInt(0), c.getString(1),  c.getString(2),  c.getString(3), c.getString(4))
            data.add(p)
            c.moveToNext()
        }
        c.close()
        return data
    }



    companion object {
        val DATABASE_NAME = "newDb"
        val DATABASE_VERSION = 1

    }
}